### DISCUSSION TITLE
Kraft XML Schema

### DISCUSSION THREAD

**Discussion by dragotin** [dragotin/kraft#98] [Ideas]
I am planning to move the document storage out of the database and save the documents as XML files in the file system. 

I created  a first draft of an example document and a xsd schema here:
https://github.com/dragotin/kraft/tree/newshared/xml

it contains a couple of new additions, such as the `itemGroup` element, that groups items together, so that we have different "chapters" on the documents.

I would be interested in feedback from people if the new format is sufficient. Thanks!

**Comment by noseshimself** [comment]
Maybe adding the Kraft and the schema version numbers is a good idea. Add a GUID that tries to be unique across all installations and objects.And if you want to include letters etc.

```
<xs:element name="itemGroup" type="itemGroup_type" minOccurs="1" maxOccurs="unbounded"/>
```

might not be true.

Defining structures for all other objects (addresses, configuration data, catalogs, catalog entries...) should be the next step there and objects for label.* (that might permit adding private labels if a user wants to) might be nice, too:

```
<kraftlabel schemaversion="1">
  <objID>G-U-I-D</objID>
  <tag>VERSION</tag>
  <comment>this label is absolutely useless</comment>
  <entry>
    <locale>en_US</locale>
    <value>Kraft version</value>
  </entry>
  <entry>
    <locale>de_DE</locale>
    <value>Kraft Version</value>
  </entry>
</kraftlabel>
```

In the end a primitive key, value list (or database) will do (redis anyone?) and the index can be rebuilt from objIDs and references if the database is getting lost as it is just accelerating things. Even not having a common database and synchronization using regular file system sync will do unless redis is also used for locking and synchronization.
